internal/guest/bridge/publisher.go

```diff
@@ -0,0 +1,68 @@
+//go:build linux
+
+package bridge
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Microsoft/hcsshim/internal/guest/prot"
+	log "github.com/Microsoft/hcsshim/internal/log"
+	eventstypes "github.com/containerd/containerd/api/events"
+	"github.com/containerd/containerd/events"
+	"github.com/containerd/containerd/runtime"
+	"github.com/containerd/containerd/runtime/v2/shim"
+	"github.com/pkg/errors"
+)
+
+const emptyActivityID = "00000000-0000-0000-0000-000000000000"
+
+// wraps `bridge.PublishNotification` to conform to `shim.Publish`, so that it can
+// be used by containerd's `oom.Watcher` (github.com/containerd/containerd/pkg/oom)
+type BridgePublisherFunc func(*prot.ContainerNotification)
```

Comment:
I feel like this is a tad confusing. Is there a reason to do this specifically like this? could we do something like the below instead? 

```
type BridgePublisher struct {
    publisherFunc func(*prot.ContainerNotification)
}

func New(publisherFunc *prot.ContainerNotificiation) BridgePublisher {
    return BridgePublisher{publishFunc} 
}

var _ shim.Publisher = BridgePublisher{} 

func (bp BridgePublisher) Close() error {
    return nil 
}

func (bp BridgePublisher) Publish(ctx context.Context, topic string, event events.Event) error {
    n, err := mapEventToNotification(topic, event)
    if err != nil {
        return errors.Wrapf(err, "could not handle topic %q and event %+v", topic, event)
    }
    bp.publisherFunc(n)
    return nil 
}
```

Reply:
none, I'll change it to a struct instead to make it clear